docs/api/role-management/delete.asciidoc

```diff
@@ -0,0 +1,14 @@
+[[role-management-api-delete]]
+===== Delete Role
+
+To delete a role in {es}'s `native` realm, submit a DELETE request to the `/api/security/role/<rolename>`
+endpoint:
+
+[source,js]
+--------------------------------------------------
+DELETE /api/security/role/my_admin_role
+--------------------------------------------------
+// KIBANA
+
+If the role is successfully deleted, the response code is `204`; otherwise, the response
+code is 404.
```

Comment:
Here are the edits I made:
--Made "Delete role" a higher level had
--Added headings for Response and Request, common in API docs 
--Added experimental tag because its its own section
--Added an intro description
--Removed the terminology "in {es}'s `native` realm" throughout to remove potential confusion with the Elasticsearch API

[[role-management-api-delete]]
=== Delete role
experimental[This API is *experimental* and may be changed or removed completely in a future release. Although the underlying mechanism of enforcing role-based access control is stable, the APIs for managing the roles are currently experimental.]

Permanently removes a Kibana role. Once you delete a role, you cannot recover it.

==== Request

To delete a role, submit a DELETE request to the `/api/security/role/<rolename>`
endpoint.

````
[source,js]
--------------------------------------------------
DELETE /api/security/role/my_admin_role
--------------------------------------------------
// KIBANA
````

==== Response

If the role is successfully deleted, the response code is `204`; otherwise, the response
code is 404.